action/protocol/staking/protocol.go

```diff
@@ -255,25 +251,10 @@ func (p *Protocol) Validate(ctx context.Context, act action.Action) error {
 
 // ActiveCandidates returns all active candidates in candidate center
 func (p *Protocol) ActiveCandidates(ctx context.Context, height uint64) (state.CandidateList, error) {
```

Comment:
should this `ActiveCandidates` take in a sr?

Reply:
exactly, then we can remove p.sr